x-pack/plugins/painless_lab/public/application/context.tsx

```diff
@@ -8,50 +8,89 @@ import React, { createContext, ReactNode, useState, useContext } from 'react';
 import { HttpSetup } from 'src/core/public';
 
 import { Links } from '../links';
+import { exampleScript, painlessContextOptions } from './constants';
+import { Store, Payload, Validation } from './types';
 
-import { initialState, Store } from './store';
-import { PAINLESS_LAB_KEY } from './constants';
+interface AppContextProviderArgs {
+  children: ReactNode;
+  value: {
+    http: HttpSetup;
+    links: Links;
+  };
+}
 
 interface ContextValue {
-  state: Store;
-  updateState: (nextState: (s: Store) => Partial<Store>) => void;
+  store: Store;
+  updatePayload: (changes: Partial<Payload>) => void;
   services: {
     http: HttpSetup;
   };
   links: Links;
 }
 
+const initialPayload = {
+  context: painlessContextOptions[0].value,
+  code: exampleScript,
+  parameters: `{
+  "string-parameter": "yay",
```

Comment:
Should we clean these up?

Reply:
Or are they intended as the parameter defaults? `"yay"` seems a bit too informal to me 😄

Reply:
Haha yes I intended them to be defaults. I'll try to find something appropriately solemn. :)